generic-server/src/plugin/container.rs

```diff
@@ -89,6 +88,35 @@ impl<'a> PluginContainer<'a> {
         }
     }
 
+    pub fn unload(&mut self) -> Result<(), PluginContainerError> {
+        // Unmount plugins and clearing the vector of routes
+        let errors: HashMap<_, _> = self
+            .plugins
+            .iter()
+            .filter_map(|plugins| match plugins.unmount() {
+                Ok(_) => {
```

Comment:
The lambda function acts as an iterator, it goes through the plugins one by one. You should then name the variable `plugin`, not `plugins`.

```suggestion
            .iter()
            .filter_map(|plugin| match plugin.unmount() {
                Ok(_) => {
```